File: fitrbackend/src/main/java/com/example/fitrbackend/dto/CreateUserRequest.java
package com.example.fitrbackend.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.constraints.NotBlank;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class CreateUserRequest {

    @NotBlank(message = "email is required")
    @JsonProperty("email")
    private String email;

    @NotBlank(message = "password is required")
    @JsonProperty("password")
    private String password;

    @NotBlank(message = "first name is required")
    @JsonProperty("firstname")
    private String firstName;

    @NotBlank(message = "last name is required")
    @JsonProperty("lastname")
    private String lastName;
}

File: fitrfrontend/fitrfrontend/Models/DTO/CreateUserRequest.swift
//
//  CreateUserRequest.swift
//  fitrfrontend
//
//  Created by Ambrose Blay on 2/17/26.
//

import Foundation

struct CreateUserRequest: Codable {
    let email: String
    let password: String
    let firstname: String
    let lastname: String
}
